tools/pipelines/test-perf-benchmarks.yml

```diff
@@ -376,351 +384,218 @@ stages:
+            inputs:
+              # It seems there's a bug and preferTriggeringPipeline is not respected.
+              # We force the behavior by explicitly specifying:
+              # - buildVersionToDownload: specific
+              # - buildId: <the id of the triggering build>
+              # preferTriggeringPipeline: true
+              source: specific
+              project: internal
+              pipeline: ${{ variables.artifactPipeline }}
+              buildVersionToDownload: specific
+              buildId: ${{ variables.artifactBuildId }}
+              artifact: pack
+              patterns: "**/${{ variables.escapedTestPackage }}-?.?.?-*.tgz"
+              path: $(Pipeline.Workspace)/client/pack
+              # allowPartiallySucceededBuilds: true # No effect as long as we have buildVersionToDownload: specific
+              # branchName: $(Build.SourceBranch)   # No effect as long as we have buildVersionToDownload: specific
 
-              # Run memory tests
-              npm run test:memory-profiling:report;
-              memoryTestsExitCode=$?;
+          # Install e2e test package
+          - task: Bash@3
+            displayName: Install Test Package
+            inputs:
+              targetType: 'inline'
+              workingDirectory: ${{ variables.testWorkspace }}
+              script: |
+                echo "Install ${{ variables.testPackage }}"
+                TEST_PACKAGE_FILE_PATTERN=${{ variables.escapedTestPackage }}-?.?.?-*.tgz
+                TEST_PACKAGE_PATH_PATTERN=$(Pipeline.Workspace)/client/pack/scoped/$TEST_PACKAGE_FILE_PATTERN
 
-              if [[ $executionTimeTestsExitCode -ne 0 ]]; then
-                echo "##vso[task.logissue type=error]Exit code for runtime tests execution = $executionTimeTestsExitCode"
-              fi
+                echo $TEST_PACKAGE_FILE_PATTERN
+                echo $TEST_PACKAGE_PATH_PATTERN
 
-              if [[ $memoryTestsExitCode -ne 0 ]]; then
-                echo "##vso[task.logissue type=error]Exit code for memory tests execution = $memoryTestsExitCode"
-              fi
+                echo `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l`
+                echo `ls $TEST_PACKAGE_PATH_PATTERN`
 
-              if [[ $executionTimeTestsExitCode -ne 0 ]] || [[ $memoryTestsExitCode -ne 0 ]]; then
-                exit 1;
-              fi
-          env:
-            FLUID_TEST_LOGGER_PKG_PATH: ${{ variables.testWorkspace }}/node_modules/@ff-internal/aria-logger # Contains getTestLogger impl to inject
-            FLUID_BUILD_ID: $(Build.BuildId)
+                if [[ `ls -1 $TEST_PACKAGE_PATH_PATTERN | wc -l` -eq 1 ]]; then
+                  TEST_PACKAGE_TGZ=`ls $TEST_PACKAGE_PATH_PATTERN`
+                else
+                  ls -1 $TEST_PACKAGE_PATH_PATTERN
+                  echo "##vso[task.logissue type=error]Test package '${{ variables.testPackage }}' not found, or there are more then one found"
+                  exit -1
+                fi
```

Comment:
This may be fine, but typically using the output of ls in this way is [discouraged](http://mywiki.wooledge.org/ParsingLs); [more info](https://unix.stackexchange.com/a/129120). `find` is better.

Oh looked closer and saw this is counting files; may be OK.